### PR TITLE
Automatically add tags to variants; fixes #230

### LINKED ISSUE #230: std::variant serialization of empty types can be irreversible
the variant's current alternative type is not serialized, making it impossible to recover when deserializing
Making this sort of use case where tagged unions are sometimes left empty unusable.

There are however no compile errors, so i believe this should either be supported or throw a compiler error.

```cpp
    struct button_pressed_t
    {
        bool operator==(const button_pressed_t&) const = default;
    };

    struct button_released_t
    {
        bool operator==(const button_released_t&) const = default;
    };

    struct key_pressed_t
    {
        char key;

        bool operator==(const key_pressed_t&) const = default;
    };

    using my_event_type_t = std::variant<button_pressed_t, button_released_t, key_pressed_t>;
    struct my_event_t
    {
        my_event_type_t event;

        [[nodiscard]] constexpr bool operator==(const my_event_t& other) const noexcept = default;
    };

    auto my_event = my_event_t{.event = key_pressed_t{}};

    auto as_json = rfl::json::write(my_event, rfl::json::pretty);
    auto reparsed_event = rfl::json::read<my_event_t>(as_json).value();

    // Should be equal
    ASSERT_NE(my_event, reparsed_event);
```

### PR DESCRIPTION
**Pull request by liuzicheng1987**

### CODE REVIEW

File: docs/processors.md [hunk 1]
```diff
@@ -61,6 +62,58 @@ The resulting JSON string looks like this:
 {"type":"Person","first_name":"Homer","last_name":"Simpson","age":45}
 ```
 
+### `rfl::AddStructsToVariants`
```
Comment (bryceschober) [inline docs/processors.md:65]: Typo: `AddStructsToVariants` -> `AddTagsToVariants`

File: docs/processors.md [hunk 2]
```diff
@@ -61,6 +62,58 @@ The resulting JSON string looks like this:
 {"type":"Person","first_name":"Homer","last_name":"Simpson","age":45}
 ```
 
+### `rfl::AddStructsToVariants` 
+
+This processor automatically adds structs to variants. Consider the following example:
```
Comment (bryceschober) [inline docs/processors.md:67]: Suggestion: "automatically adds struct tags to"

File: docs/processors.md (hunk 1 above)
Comment (liuzicheng1987) [reply docs/processors.md:65]: Thanks!

File: docs/processors.md (hunk 2 above)
Comment (liuzicheng1987) [reply docs/processors.md:67]: That was meant to be "tags" as well...

**Pull request merged**